### PR TITLE
Add plans doc

### LINKED ISSUE #242: Custom bootstrap local plan
New section about providing custom system agent plans that are executed at bootstrap phase.

### PR DESCRIPTION
**Pull request by Itxaka**
Fixes #242 

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: docs/elemental-plans.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+## Introduction
+
+Elemental uses the [Rancher System Agent](https://github.com/rancher/system-agent) renamed to Elemental System Agent,
+to initially bootstrap the node with a simple plan that sets some labels for the node,
+sets the proper hostname according to the `#!yaml MachineInventory`
+and installs the default Rancher System Agent from Rancher Server which then will install the proper Kubernetes components.
+
+This small bootstrap service also accepts local plans stored under `/var/lib/elemental/agent/plans` so any plan written
+in there will also be applied during the initial node start (After it has been installed).
```
Comment (nunix) [inline docs/elemental-plans.md:9]:
```suggestion
The bootstrap service also accepts local plans stored under `/var/lib/elemental/agent/plans`. Any plan written
in there will also be applied during the initial node start after the installation is completed.
```

File: docs/elemental-plans.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+!!! note
+    This local plans run only during the initial Elemental bootstrap **before** Kubernetes is installed on the node
```
Comment (nunix) [inline docs/elemental-plans.md:12]:
```suggestion
    The local plans run only during the initial Elemental bootstrap **before** Kubernetes is installed on the node.
```

File: docs/elemental-plans.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        debug: true
+  machineName: my-machine
+  machineInventoryLabels:
+    location: "europe"
+```
+
+
+## Plan examples
+
+These plans are provided as a quick reference and not guaranteed to work. To learn more about plans please check [Rancher System Agent](https://github.com/rancher/system-agent)
```
Comment (nunix) [inline docs/elemental-plans.md:73]:
```suggestion
The following plans are provided as a quick reference and are not guaranteed to work in your environment. To learn more about plans please check [Rancher System Agent](https://github.com/rancher/system-agent).
```

File: docs/elemental-plans.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
```
Comment (nunix) [inline docs/elemental-plans.md:74]:
```suggestion

**Example 1: one time instructions**

```

File: docs/elemental-plans.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        },
+        {
+            "name":"stop sshd service",
+            "command":"systemctl",
+            "args": ["stop", "sshd"]
+        }
+    ]
+}
+```
+
```
Comment (nunix) [inline docs/elemental-plans.md:91]:
```suggestion

**Example 2: periodic instructions**
```

File: docs/elemental-plans.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    [
+        {
+            "name":"set hostname",
+            "image":"ghcr.io/rancher-sandbox/elemental-example-plan:main"
+            "command": "run.sh"
+        }
+    ]
+}
+```
+
```
Comment (nunix) [inline docs/elemental-plans.md:103]:
```suggestion

Example 3: files
```

File: docs/elemental-plans.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    [
+        {
+            "content":"Welcome to the system",
+            "path":"/etc/motd",
+            "permissions": "0644"
+        }
+    ]
+}
+```
+
```
Comment (nunix) [inline docs/elemental-plans.md:115]:
```suggestion

Example 4: probes
```

**Review by nunix — COMMENTED** [review]
Added introduction sentences for the examples and rephrased introduction.

**Review by mjura — APPROVED** [review]

**Comment by Itxaka** [comment]
@nunix I added most of your comments except the examples header as I changed it to be an example block AND collapsed so they dont take the whole page and can be clicked to expand

**Review by nunix — APPROVED** [review]

**Pull request merged**